backend/src/helpers/students.js

```diff
@@ -52,3 +53,18 @@ exports.getStatusByStudentId = async function (studentID) {
     return { err }
   }
 }
+
+exports.getStatusPromise = async function (body) {
+  // Check if request contains StatusID
+  if (body.StatusID == null) {
+    if (body.StatusString != null) {
+      const statusString = body.StatusString
+      delete body.StatusString
```

Comment:
I personally think this should happen in the code that calls `getStatusPromise`. It's possible that someone uses this helper without expecting the deletion to occur.

Reply:
yes, we have refactored the function. `getStatusPromise` is a pure function now that only takes in the `statusString` and returns the `status`